crates/subspace-farmer/src/dsn.rs

```diff
@@ -170,18 +172,23 @@ impl DSNSync for subspace_networking::Node {
                         tracing::debug!(?peers, "Received peers");
 
                         // select first peer for the piece-by-range protocol
-                        for id in peers.into_iter().take(5) {
-                            if node
-                                .send_generic_request(id, PeerInfoRequest)
-                                .await
-                                .map(|response| response.peer_info.status == PeerSyncStatus::Ready)
-                                .unwrap_or_default()
-                            {
-                                return Ok(id);
+                        for id in peers {
+                            match node.send_generic_request(id, PeerInfoRequest).await {
+                                Ok(PeerInfoResponse {
+                                    peer_info: PeerInfo { status },
+                                }) if status == PeerSyncStatus::Ready => return Ok(id),
+                                Ok(PeerInfoResponse {
+                                    peer_info: PeerInfo { status },
+                                }) => trace!(%id, ?status, "Peer is not ready for synchronization"),
```

Comment:
I really dislike `if` in matches, it makes match statements heavier to read. If inside would be much easier to parse.